File: src/main/java/com/github/brankale/dto/GamePageDto.java
package com.github.brankale.dto;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public record GamePageDto(
        @SerializedName("@context") String context,
        @SerializedName("@type") String type,
        @SerializedName("name") String name,
        @SerializedName("image") List<String> images,
        @SerializedName("description") String description,
        @SerializedName("sku") Long sku,
        @SerializedName("category") String category,
        @SerializedName("releaseDate") String releaseDate,
        @SerializedName("gtin13") String gtin13,
        @SerializedName("brand") BrandDto brand,
        @SerializedName("offers") List<OffersDto> offers,
        @SerializedName("aggregateRating") RatingDto aggregateRating
) {
//    @Override
//    public String toString() {
//        return "GamePageDto{" +
//                "context='" + context + '\'' +
//                ", type='" + type + '\'' +
//                ", name='" + name + '\'' +
//                ", image=" + images +
//                ", description='" + description + '\'' +
//                ", sku=" + sku +
//                ", category='" + category + '\'' +
//                ", releaseDate='" + releaseDate + '\'' +
//                ", gtin13='" + gtin13 + '\'' +
//                ", brand=" + brand +
//                ", offers=" + offers +
//                ", aggregateRating=" + aggregateRating +
//                '}';
//    }
}

File: src/main/kotlin/com/github/brankale/parsers/game_page/dto/GamePageDto.kt
package com.github.brankale.parsers.game_page.dto

import com.google.gson.annotations.SerializedName

data class GamePageDto(
        @SerializedName("@context") val context: String,
        @SerializedName("@type") val type: String,
        @SerializedName("name") val name: String,
        @SerializedName("image") val images: List<String>,
        @SerializedName("description") val description: String,
        @SerializedName("sku") val sku: Long,
        @SerializedName("category") val category: String,
        @SerializedName("releaseDate") val releaseDate: String,
        @SerializedName("gtin13") val gtin13: String,
        @SerializedName("brand") val brand: BrandDto,
        @SerializedName("offers")  val offers: List<OffersDto>,
        @SerializedName("aggregateRating")  val aggregateRating: RatingDto
)
